controllers/users/authenticate-web.js

```diff
@@ -21,26 +22,44 @@ const authenticateWeb = async (req, res) => {
   /**
    * @type {RegisterBody}
    */
-  const users = req.body;
+  const {exchangeToken} = req.body;
 
-  const checkUser = await User.findOne({
-    where: {
-      human_id: users.human_id
-    }
-  });
+  const verifyTokenResponse = await HumanIdService.verifyExchangeToken(exchangeToken);
+  if (!verifyTokenResponse.success) {
+    return res.status(500).json({
+      status: 'error on verify exchange token',
+      code: 500
+    });
+  }
+
+  const {data} = verifyTokenResponse || {};
+  const {appUserId} = data || {};
+
+  if (!appUserId) {
+    return res.status(500).json({
+      status: 'App user id is required',
+      code: 500
+    });
+  }
+
+  const checkUser = await UsersFunction.findUserByHumanId(User, appUserId);
 
   let insertedObject = {};
 
   if (checkUser) {
     insertedObject = {
       user: checkUser,
-      anonymousUser: await UsersFunction.findAnonymousUserId(User, checkUser.dataValues.user_id, {
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, checkUser.user_id, {
         raw: false
       })
     };
   } else {
-    users.username = 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS');
-    users.verified_status = 'UNVERIFIED';
+    let users = {
+      username: 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS'),
+      verified_status: 'UNVERIFIED',
+      human_id: appUserId,
+      country_code: 'US'
+    };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for checking if a user exists and creating one if they don't is sound. However, the username generation logic uses `moment`, which has been removed from the imports. This will cause a runtime error. You need to replace it with a suitable alternative like `Date`. Also, the hardcoded 'US' country code might not be appropriate for all users. If possible, consider retrieving this information from the user's profile or request data.
```diff
-      username: 'auth_web_' + moment().format('YYYYMMDDHHmmssSSS'),
+      username: 'auth_web_' + Date.now(),
...
-      country_code: 'US'
+      country_code: // retrieve from user's profile or request data
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->